src/Executables/PrintMinimumGridSpacing/PrintMinimumGridSpacing.cpp

```diff
@@ -0,0 +1,154 @@
+  };
+  using options = tmpl::list<Value>;
+  static constexpr OptionString help = {
+      "Defines a number of dimensions (1, 2, or 3)"};
+  DimensionOption() = default;
+  DimensionOption(size_t value, const OptionContext& context)
+      : value_(std::move(value)) {
+    if (value_ != 1 and value_ != 2 and value_ != 3) {
+      PARSE_ERROR(context, "Dimemsion Value must be 1, 2, or 3 but is "
+                               << value_);
+    }
+  }
+  size_t value_{};
+};
+
+template <size_t Dim>
+void compute_and_print_minimum_grid_spacing(std::string input_file) {
+  Options<tmpl::list<Dimension, OptionTags::DomainCreator<Dim, frame>>> options(
+      "");
+  options.parse_file(input_file);
+
+  const auto domain_creator =
+      options.template get<OptionTags::DomainCreator<Dim, frame>>();
+  auto domain = domain_creator->create_domain();
+  double min_grid_spacing = std::numeric_limits<double>::max();
+  for (const auto& block : domain.blocks()) {
+    const auto initial_ref_levs =
+        domain_creator->initial_refinement_levels()[block.id()];
+    const std::vector<ElementId<Dim>> element_ids =
+        initial_element_ids(block.id(), initial_ref_levs);
+    for (const auto& element_id : element_ids) {
+      ElementMap<Dim, frame> map(element_id,
+                                 block.coordinate_map().get_clone());
+      Mesh<Dim> mesh(domain_creator->initial_extents()[block.id()],
+                     Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto);
+      min_grid_spacing = std::min(
+          min_grid_spacing,
+          minimum_grid_spacing(mesh.extents(), map(logical_coordinates(mesh))));
+    }
+  }
+  Parallel::printf("The minimum grid spacing is: %1.14e\n", min_grid_spacing);
+}
+
+std::string custom_help_msg() {
+  constexpr int max_label_size = 21;
+  const std::string help_text =
+      "Print the minimum grid spacing between inertial coordinates of the "
+      "Domain specified in the input file. The output can be used to "
+      "choose appropriate time steps.";
+
+  std::ostringstream ss;
+  ss << "\n==== Description of expected options:\n" << help_text
+     << "\n\nOptions:\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << "Dimension"
+     << "Value\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << ""
+     << "The dimensions of the desired domain (1, 2, or 3).\n\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << "DomainCreator"
+     << "DomainCreator<Dim, Frame::Inertial>\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << ""
+     << "The domain to create initially.\n"
+     << "  " << std::setw(max_label_size + 2) << std::left << ""
+     << "Dim must be equal to Dimension's Value.\n\n";
+
+  return ss.str();
+}
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
```

Comment:
Could you move the `CkRegisterMainModule` function to the top of the file, right below the includes?